scripts/generate.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+[[ "${1:-}" != "--help" ]] || {
+  cat <<-EOF
+	Gets .did files and generates canister bindings from them.
+
+	Note: This does so WITHOUT deploying any canisters.  Typically "dfx generate"
+	requires all the canisters to be deployed locally, which seems absurdly heavyweight.
+	This code just downloads the candid files and puts them where they would normally
+	be found for a local deployment.  Much faster!
+	EOF
+  exit 0
+}
+
+# Gets all .did files listed in dfx.json.
+#
+# .did files are placed in the local .dfx directory, where the .did files are expected by `dfx generate` and other commands.
+function install_did_files() {
+  jq -r '.canisters | to_entries | .[] | select(.value.candid != null) | "\(.key) \(.value.candid)"' dfx.json |
```

Comment:
should we add a check that assert if `jq` is installed at the top of the script? "You don't have jq, install jq to execute this script"

Reply:
Yes, can do.  Added to the `--help` and added a check.